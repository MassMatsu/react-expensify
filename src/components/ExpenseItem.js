import React from 'react'

const ExpenseItem = ({id, description, amount, createdAt}) => {
  return (
    <div>
    <h3>{description} </h3>
      <p>{amount} - {createdAt}</p>
    </div>
  )
}

export default ExpenseItem