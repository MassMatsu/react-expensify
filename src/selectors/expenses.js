// function for all filtering at once by taking expenses state and fitlers state

const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses
    .filter((expense) => {
      const startDateMatch =
        startDate === undefined || expense.createdAt >= startDate;
      const endDateMatch =
        endDate === undefined || expense.createdAt <= endDate;
      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase()); // if text = '', includes method always returns 'true'
      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      if (sortBy === 'date') {
        return a.createdAt < b.createdAt ? 1 : -1;
      } else if (sortBy === 'amount') {
        return a.amount < b.amount ? 1 : -1;
      }
    });
};

export default getVisibleExpenses;
