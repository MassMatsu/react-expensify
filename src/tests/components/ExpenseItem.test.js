import React from 'react'
import {shallow} from 'enzyme'
import ExpenseItem from '../../components/ExpenseItem'
import expenses from '../fixtures/expenses'

test('should render ExpenseItem correctly', () => {
  const wrapper = shallow(<ExpenseItem {...expenses[0]}/>)
  expect(wrapper).toMatchSnapshot()
})
