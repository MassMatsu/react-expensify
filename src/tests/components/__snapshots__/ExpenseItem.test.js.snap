// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseItem correctly 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      rent
       
    </h3>
  </Link>
  <p>
    23499
     - 
    0
  </p>
</div>
`;
